cmake/sanitizer.cmake

```diff
@@ -0,0 +1,41 @@
+#if ((NOT ${CMAKE_CXX_COMPILER_ID} MATCHES "Clang") AND (NOT ${CMAKE_CXX_COMPILER_ID} MATCHES "GNU"))
```

Comment:
Is it necessary here? Or we can remove it?

Reply:
I think I'll extend it to include Mac OS compiler and uncomment it